libretro/libretro-core.c

```diff
@@ -1912,10 +1877,69 @@ static void update_variables(void)
 
       RETROC64MODL=modl;
       if (retro_ui_finalized)
-        c64model_set(modl);
+      {
+         c64model_set(modl);
+
+         int eng=0,modl=0,sidmdl=0;
+         resources_get_int("SidEngine",&eng);
+         resources_get_int("SidModel",&modl);
+         sidmdl=((eng << 8) | modl);
+         resid_model_changed = (RETROSIDMODL != sidmdl);
+      }
    }
 #endif
 
+   var.key = "vice_sid_model";
+   var.value = NULL;
+
+   if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
+   {
+      int eng=0,modl=0,sidmdl=0;
+
+      if (strcmp(var.value, "6581F") == 0) { eng=0; modl=0; }
+      else if (strcmp(var.value, "8580F") == 0) { eng=0; modl=1; }
+      else if (strcmp(var.value, "6581R") == 0) { eng=1; modl=0; }
+      else if (strcmp(var.value, "8580R") == 0) { eng=1; modl=1; }
+      else if (strcmp(var.value, "8580RD") == 0) { eng=1; modl=2; }
+      else if (strcmp(var.value, "DefaultF") == 0) { eng=0; modl=-1; }
+      else if (strcmp(var.value, "DefaultR") == 0) { eng=1; modl=-1; }
+
+      if (modl<0)
+         if (retro_ui_finalized)
+            resources_get_int("SidModel",&modl);
+         else
+            modl=1;
+
+      sidmdl=((eng << 8) | modl);
+      if (retro_ui_finalized)
+      {
+         if (RETROSIDMODL != sidmdl || resid_model_changed)

```

Comment:
This variable resid_model_changed doesn’t exist here if _Vic20_ is defined. Please test building all cores and fix.

Reply:
Will be fixed with proper guard.